### PR TITLE
skip_files

### PR DESCRIPTION
**Pull request by yorinasub17**
This introduces `skip_files` to boilerplate templates, which allow you to specify files that boilerplate should conditionally ignore. This is useful for boilerplate documentation files, as well as creating templates that conditionally render different snippets based on user input (e.g., data template files for the architecture catalog).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -488,6 +502,52 @@ Note the following:
...
+#### Skip Files
+
+You can specify files that should be excluded from the rendered output using the `skip_files` section in
+`boilerplate.yml`. This is most useful when you have templates that need to conditionally exclude files from the
+rendered folder list.
+
+The `skip_files` section is a list of objects with the fields `path` and `if`, where `path` is required. `if` can be
+used to conditionally skip a file from the template folder, and it defaults to `true`. That is, when `if` is omitted,
+the file at the path is always excluded from the output. Note that `path` is always the relative path from the template
+root. Both `path` and `if` support Go templating syntax with access to boilerplate [variables](#variables) and [template
```
Comment (brikis98) [inline README.md:514]: Does `path` support wildcards / globs?

File: templates/template_processor.go [hunk 2]
```diff
@@ -219,6 +219,9 @@ func cloneOptionsForDependency(dependency variables.Dependency, originalOpts *op
 	}
 
 	templateUrl, templateFolder, err := options.DetermineTemplateConfig(renderedTemplateUrl)
+	if err != nil {
```
Comment (brikis98) [inline templates/template_processor.go:222]: Doh.

File: templates/template_processor.go [hunk 3]
```diff
@@ -295,13 +298,33 @@ func shouldSkipDependency(dependency variables.Dependency, opts *options.Boilerp
 	return rendered == "true", nil
 }
 
+// Return true if the if parameter of the given SkipFile evaluates to a "true" value
+func skipFileIfCondition(skipFile variables.SkipFile, opts *options.BoilerplateOptions, variables map[string]interface{}) (bool, error) {
+	if skipFile.If == "" {
+		return false, nil
+	}
```
Comment (brikis98) [inline templates/template_processor.go]: I thought if you didn't specify `if` it would default to `true`?

File: variables/skip_files.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	// unmarshalStringField only returns nil pointer if there is an error, so we can assume it is not nil here.
+	path := *pathPtr
+
+	skipIfPtr, err := unmarshalStringField(fields, "if", false, path)
+	if err != nil {
+		return nil, err
+	}
+	// If the `if` attribute is omitted, we default it to true.
```
Comment (brikis98) [inline variables/skip_files.go]: Ohh... You default it to the string "true" in the parsing logic... Hm, I find that slightly unexpected TBH. It's a NIT, but for some reason, the "principle of least" surprise tells me that the parser should return `if` as empty string if the user didn't specify it, and the logic checking for the value of `if` treats empty string as a signal to default to a boolean `true`.

Comment (yorinasub17) [reply variables/skip_files.go]: Ok. I don't feel strongly about this so updated to what you mentioned: 9405f4d

File: templates/template_processor.go (hunk 3 above)
Comment (yorinasub17) [reply templates/template_processor.go]: Updated in 9405f4d to what you suggest down below.

File: README.md (hunk 1 above)
Comment (yorinasub17) [reply README.md:514]: Implemented glob support 85658d4

**Pull request merged**